### PR TITLE
prometheus pushgateway support

### LINKED ISSUE #17: Store regression results to prometheus
Currently tests results are stored only in CSV files. Need to provide option to store results in prometheus also, as it implemented in https://github.com/bblfsh/performance

### PR DESCRIPTION
**Pull request by lwsanty**
Add prometheus pushgateway support to work on the same principle as --csv flag.

closes #17

Signed-off-by: lwsanty <lwsanty@gmail.com>

### CODE REVIEW

**Comment by lwsanty** [comment]
**TODO** after this issue is merged we should change `Jenkinsfile` inside `gitbase` repo

File: test.go [hunk 1]
```diff
@@ -203,6 +207,18 @@ func (t *Test) SaveLatestCSV() {
 	}
 }
 
+// StoreLatestToPrometheus stores latest version results to prometheus pushgateway
+func (t *Test) StoreLatestToPrometheus() {
```
Comment (erizocosmico) [inline test.go]: If promConfig and ciConfig are only used here, I would consider not adding them to the Test struct and instead pass them as parameters to this function

File: test.go [hunk 2]
```diff
@@ -203,6 +207,18 @@ func (t *Test) SaveLatestCSV() {
...
+	version := t.config.Versions[len(t.config.Versions)-1]
+	cli := NewPromClient(t.promConfig)
+	for _, q := range t.queries {
+		res := average(t.results[version][q.ID])
+		if err := cli.Dump(res, version, q.ID, t.ciConfig.Branch, t.ciConfig.Commit); err != nil {
+			panic(err)
```
Comment (erizocosmico) [inline test.go]: I would return an error instead of panicking here. Then if needed panic in the main

**Review by erizocosmico — CHANGES_REQUESTED** [review]

File: test.go (hunk 1 above)
Comment (lwsanty) [reply test.go]: Done

File: test.go (hunk 2 above)
Comment (lwsanty) [reply test.go]: Done

File: test_test.go [hunk 3]
```diff
@@ -17,7 +17,7 @@ func TestTest(t *testing.T) {
 	config.Complexity = 0
 	config.Repeat = 1
 
-	test, err := NewTest(config)
+	test, err := NewTest(config, PromConfig{}, CIConfig{})
```
Comment (erizocosmico) [inline test_test.go]: this is making tests fail, constructor no longer takes prometheus and ci config

**Review by erizocosmico — CHANGES_REQUESTED** [review]

File: test_test.go (hunk 3 above)
Comment (lwsanty) [reply test_test.go]: ouch, typo, wait a sec

Comment (lwsanty) [reply test_test.go]: Done

**Review by erizocosmico — APPROVED** [review]

**Review by juanjux — APPROVED** [review]

**Review by ajnavarro — APPROVED** [review]

**Pull request merged**